gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/MongoPortalPageContextRepository.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.repository.mongodb.management;
+
+import io.gravitee.repository.exceptions.TechnicalException;
+import io.gravitee.repository.management.api.PortalPageContextRepository;
+import io.gravitee.repository.management.model.PortalPageContext;
+import io.gravitee.repository.management.model.PortalPageContextType;
+import io.gravitee.repository.mongodb.management.internal.model.PortalPageContextMongo;
+import io.gravitee.repository.mongodb.management.internal.portalpagecontext.PortalPageContextMongoRepository;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author GraviteeSource Team
+ */
+@Component
+@Slf4j
+public class MongoPortalPageContextRepository implements PortalPageContextRepository {
+
+    @Autowired
+    private PortalPageContextMongoRepository internalRepo;
+
+    @Override
+    public Optional<PortalPageContext> findById(String id) throws TechnicalException {
+        log.debug("Find PortalPageContext by id [{}]", id);
+        Optional<PortalPageContextMongo> portalPageContextMongo = internalRepo.findById(id);
```

Comment:
```suggestion
        Optional<PortalPageContext> result = internalRepo.findById(id).map(this::map);
    log.debug("Find PortalPageContext by id [{}] - {}", id, result.isPresent() ? "Found" : "Not found");
    return result;
```

Don't know if it's clearer but you can chain the optionals without creating 2 variables here I think. Wdyt ?

Reply:
Done